BioForm/profiles/static/scripts/register.js

```diff
@@ -0,0 +1,158 @@
+const form = document.getElementById("form");
+const username = document.getElementById("username");
+const first_name = document.getElementById("first-name");
+const last_name = document.getElementById("last-name");
+const email = document.getElementById("email");
+const password1 = document.getElementById("password");
+const password2 = document.getElementById("password-confirmation");
+const submit = document.getElementById("submit")
+
+// results object used to check if all fields are validated
+results = {
+    "username": false,
+    "first_name": false,
+    "last_name": false,
+    "email": false,
+    "passwords": false
+};
+
+
+function validate_username(){
+    // In case there was already an error displayed
+    clear_error(username);
+    const username_value = username.value.trim();
+    results['username'] = is_valid_username(username_value);
+    unlock_or_lock_submit()
+}
+
+
+// Clears any errors that are displayed on a field already
+function clear_error(input){
+    small_element = input.parentElement.querySelector("small")
+    if(small_element)
+        small_element.remove();
+}
+
+
+function is_valid_username(username_value){
+    const username_regex = /^[\w-]+$/;
+    const message = "Invalid username entered. Please only use a combination of alphanumeric, _ and -";
+    return validation_for_username_or_name(username_value, username, username_regex, message);
+}
+
+
+function validation_for_username_or_name(value, input, regex, message){
+    
+    if (value.length >= 3 && value.length <= 15)
+        if(regex.test(value))
+            return true;
+        else
+            set_error_for(input, message);
+    else
+        set_error_for(input, "The length should be 3-15 characters. please try again.");
+
+    return false;
```

Comment:
It is better to add curly braces `{}` even if a code block has only one statement, it increases readability, and extending the block becomes easy.
```suggestion
    if (value.length < 3 && value.length > 15) {
        set_error_for(input, "The length should be 3-15 characters. please try again.");
        return false;
    }

    if(!regex.test(value)) {
        set_error_for(input, message);
        return false;
    }
    
    return true;
```